pkgs/development/python-modules/keras/default.nix

```diff
@@ -19,16 +18,15 @@
   packaging,
   rich,
   tensorflow,
-  tf-keras,
+  pythonAtLeast,
+  distutils,
 }:
 
 buildPythonPackage rec {
   pname = "keras";
   version = "3.5.0";
   pyproject = true;
 
-  disabled = pythonOlder "3.9";
-
```

Comment:
I'm not a fan of blanket-removing `disabled`. In this case it is still in sync, and many other contributors do make an effort to keep it in sync.

Reply:
We have removed it in the past for versions that are way out of range of what we support in tree.

Reply:
> We have removed it in the past for versions that are way out of range of what we support in tree.

This was indeed my rationale. 3.9 being 2 versions below 3.11, I thought legit removing it.

Reply:
pypy is still on 3.9

Reply:
and what this essentially does is turn a build-time failure to an eval-time one, which when building the `python37Packages` world without aid from the nixpkgs cache would save you a lot of trouble

Reply:
Would a passthru.test that tests `disabled` against the built wheel METADATA be of interest? That would simplify the maintenance burden

Reply:
(not blocking a merge btw)

Reply:
> pypy is still on 3.9

What I meant is that the two versions that we officially support are currently 3.11 and 3.12.
Hence, I believe that removing those `disabled` statements when the supported version is lower than 3.10-11 does not impact too many people.

Anyway, I totally get the other side of the coin, preventing user triggering useless builds.
I think that in the end, the package maintainer should have the final say. As I am not the one for `tf-keras`, I can revert my change for this package if you want.